commands/addResource.ts

```diff
@@ -1,56 +1,270 @@
+  const categories = resource.category && resource.category.length > 0 ? resource.category.map(c => c.name).join(', ') : 'Category';
+  const tags = resource.tags && resource.tags.length > 0 ? resource.tags.map(t => t.name).join(', ') : 'Tags';
+
+  resourceEmbed.setFields([
+    { name: 'level', value: resource.level ?? 'Level', inline: true },
+    { name: 'mediatype', value: resource.mediaType ?? 'Media Type', inline: true },
+    { name: 'blockchain', value: blockChains, inline: false },
+    { name: 'category', value: categories, inline: true },
+    { name: 'tags', value: tags, inline: true },
+  ]);
+  return resourceEmbed;
+}
 
 export async function execute(interaction: CommandInteraction) {
-    console.log(interaction.user.discriminator)
-    if (!await isContributor(interaction.user.id)) {
-        await interaction.reply(`it looks like you are not a contributor yet!\nPlease add yourself using: ${inlineCode('/add-contributor')}`)
-        return
-    }
-
-    const userInput = interaction.options.getString('input')
-    const validAnswers = ['y', 'n']
-    const filter = (response: Message) => {
-        return validAnswers.some(answer => answer.toLowerCase() === response.content.toLowerCase()) && response.author.id === interaction.user.id
-    }
-
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    if (!isValidUrl(userInput!)) {
-        await interaction.reply({
-            content: `you are submiting ${userInput} as a new resource. \n Is this link correct? [Y/N]`,
-            ephemeral: true,
-        })
-        const collection = await interaction.channel?.awaitMessages({ filter, max: 1, time: 15000, errors: ['time'] })
-        collection?.map((message: Message) => {
-            switch (message.content.toLowerCase()) {
-                case 'y':
-                    interaction.followUp({
-                        content: '',
-                        ephemeral: true,
-                    })
-                    break;
-                case 'n':
-                    interaction.followUp({
-                        content: 'Try again with the correct resource please!',
-                        ephemeral: true,
-                    })
-                    break;
-                default:
-                    return new Error('something bad happened')
-            }
-        })
+  let contributor;
+  if (!await isContributor(interaction.user)) {
+    await interaction.reply(`it looks like you are not a contributor yet!\nPlease add yourself using: ${inlineCode('/add-contributor')}`)
+    return
+  }
+  else {
+    const contributorResponse = await findContributor(interaction.user);
+    if (contributorResponse) {
+      contributor = contributorResponse.getId();
+    }
+  }
+
+  const url = interaction.options.getString('url');
+  if (!url) {
+    interaction.reply({ content: 'URL required, please try submitting again.', ephemeral: true });
+    return;
+  }
```

Comment:
Is this just to make typescript happy? With the `isRequired` field being true I would think this check isn't necessary.

Reply:
Yeah, the return type from `options.getString()` is `string | null`